src/cpp/transport/UDPTransportInterface.cpp

```diff
@@ -398,18 +397,20 @@ bool UDPTransportInterface::Receive(UDPChannelResource* pChannelResource, octet*
     }
 }
 
-bool UDPTransportInterface::ReleaseInputChannel(const Locator_t& locator)
+bool UDPTransportInterface::ReleaseInputChannel(const Locator_t& locator, UDPChannelResource* channel)
 {
-    std::unique_lock<std::recursive_mutex> scopedLock(mInputMapMutex);
-    if (!IsInputChannelOpen(locator))
+    if (!channel->IsAlive())
     {
         return false;
     }
 
     try
     {
+        channel->Disable();
+
         ip::udp::socket socket(mService);
         socket.open(GenerateProtocol());
+        socket.bind(GenerateEndpoint("127.0.0.1", 0));
```

Comment:
How does this work on UDPv6? I think we should call a GenerateLocalhostEndpoint method or something similar.